### PR TITLE
add fillna in xshards

### PR DESCRIPTION
**Pull request by songhappy**
## Description
reuse some code of Friesian Table to support Orca Xshards fillna

### 1. Why the change?
to avoid depleted code in orca xshards and friesian table 

### 2. User API changes
no change in friesian table

### 3. Summary of the change 
move scala Friesian Table fillna related implementation to Orca since Friesian depends on Orca
move python friesian.feature.utils fillna related to orca.data.utils 
add fillna in python orca.data.shard.py

### 4. How to test?
- [ ] Unit test

### 5. New dependencies
no dependencies

### CODE REVIEW

File: python/orca/src/bigdl/orca/data/shard.py [hunk 1]
```diff
@@ -633,6 +644,19 @@ def zip(self, other):
...
+               columns: Optional[Union[str, List[str]]]):
+        if self._get_class_name() == 'pandas.core.frame.DataFrame':
+            import pandas as pd
+            df = self.to_spark_df()
+            filled_df = fill_na(df, value, columns)
+            data_shards = spark_df_to_pd_sparkxshards(filled_df)
+            return data_shards
+        else:
+            invalidInputError(False,
+                              "Currently only support dedup() on XShards of Pandas DataFrame")
```
Comment (dding3) [inline python/orca/src/bigdl/orca/data/shard.py:657]: .... only support fillna() on XShards ...?

File: python/orca/src/bigdl/orca/data/shard.py [hunk 2]
```diff
@@ -633,6 +644,19 @@ def zip(self, other):
                               "The two SparkXShards should have the same number of elements "
                               "in each partition")
 
+    def fillna(self, value: Union[int, float, str, bool],
+               columns: Optional[Union[str, List[str]]]):
```
Comment (dding3) [inline python/orca/src/bigdl/orca/data/shard.py:648]: maybe we can also add some comments to this api?

File: python/orca/src/bigdl/orca/data/shard.py [hunk 3]
```diff
@@ -633,6 +644,19 @@ def zip(self, other):
...
+        if self._get_class_name() == 'pandas.core.frame.DataFrame':
+            import pandas as pd
+            df = self.to_spark_df()
```
Comment (jason-dai) [inline python/orca/src/bigdl/orca/data/shard.py:651]: why do we need to convert to spark df? I think we can simply do a `mapPartition` and call Pandas `fillna`, yes?

Comment (songhappy) [reply python/orca/src/bigdl/orca/data/shard.py:651]: yes, the original design of this method wanted to leverage Friesian fillna with performance. but it is not necessary.  closed this PR

**Comment by songhappy** [comment]
closed due to better implementation of just calling pandas dataframe apis

**Pull request closed without merging**